src/map/unit.cpp

```diff
@@ -3329,6 +3338,7 @@ static int32 unit_attack_timer_sub(struct block_list* src, int32 tid, t_tick tic
 	if( sd && battle_config.prevent_logout_trigger&PLT_ATTACK )
 		sd->canlog_tick = gettick();
 
+	map_freeblock_unlock();
```

Comment:
The lock is applied within an if. This means you may unlock too often now.

Reply:
Hmm true, that means we need to lock earlier actually.

Reply:
The source dies from reflect damage when calling:
```
ud->attacktarget_lv = battle_weapon_attack(src,target,tick,0);
```